core/include/detray/core/detail/single_store copy.hpp

```diff
@@ -0,0 +1,160 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s)
+#include "detray/core/detail/container_views.hpp"
+#include "detray/core/detail/data_context.hpp"
+#include "detray/definitions/indexing.hpp"
+#include "detray/definitions/qualifiers.hpp"
+
+// Vecmem include(s)
+#include <vecmem/containers/data/jagged_vector_buffer.hpp>
+#include <vecmem/containers/data/jagged_vector_view.hpp>
+#include <vecmem/containers/data/vector_buffer.hpp>
+#include <vecmem/containers/data/vector_view.hpp>
+#include <vecmem/containers/device_vector.hpp>
+#include <vecmem/containers/jagged_device_vector.hpp>
+#include <vecmem/containers/jagged_vector.hpp>
+#include <vecmem/containers/vector.hpp>
+
+#include <vecmem/memory/memory_resource.hpp>
+#include <vecmem/utils/copy.hpp>
+
+// System include(s)
+#include <type_traits>
+
+namespace detray {
+
+/// @tparam T The type of the collection data, e.g. transforms
+/// @tparam context_t the context with which to retrieve the correct data.
+template <typename T, typename context_t>
+class single_store_base {
```

Comment:
What happens if ```T``` defines its own custom view and buffer types because it holds a number of vector collections itself?

Reply:
`T` holding vector collections Is even allowed?

Reply:
Ok, no you are right, I did not look at this class long enough :D and in the single store this won't happen. But we need to be able to take arbitrary collection types not just vectors of PODs